Dark mode

nbelzer opened the pull request:
Resolves #1

- Implements a dark mode.
<img width="887" alt="screenshot 2018-12-21 at 19 50 22" src="https://user-images.githubusercontent.com/7622933/50358459-b0537b00-0559-11e9-890e-18bb778fb606.png">
<img width="887" alt="screenshot 2018-12-21 at 19 50 26" src="https://user-images.githubusercontent.com/7622933/50358460-b0537b00-0559-11e9-94d2-9e3b77f06838.png">

- A toggle is used to switch between light and dark mode.
- The preference of dark mode is stored as a cookie for later sessions.
- Adds the [Feather](https://feathericons.com/) icon set.

Commit:
Move variables to their respective file

Commit:
Split up styling and structural css

By dividing the styling and structural css it will be easier to see
what elements to change when we only want to change the looks, for
example for a dark mode.

Commit:
Add Dark Mode styling

Commit:
Add Toggle for Dark Mode

The toggle is a feather icon which changes the current state from
either dark to light or light to dark.

Commit:
Add CookieHandler

To manage cookies a simple cookiehandler has been implemented.

Commit:
Add dark-mode cookie

This cookie keeps track of the users preference for dark mode. If the
user does not permit cookies the preference will not be saved for next
visit.

MateyByrd commented:
There is a slight inconsistency with link colours, likely this is due to the fact that some links can be visited while others cannot (like the dark-mode button). This is visible in both dark and light mode.

<img width="483" alt="screenshot 2018-12-28 at 13 19 45" src="https://user-images.githubusercontent.com/7622933/50515155-5431af00-0aa3-11e9-8953-0e1db2da6a5d.png">

Commit:
Move link styling to main sass file

Commit:
Override :visited styling for menu items

Commit:
Change transitions to ease-out timing

Commit:
Improve menu baseline styling

Commit:
Use Github icon for menu item

Replaced the menu item "GITHUB" with the Github icon provided by
Feather Icons set.

MateyByrd commented:
- Updated the link styling to make sure the menu items do not change their colour when `:visited` for both dark- and light mode.
- Change the `GITHUB` menu item to an icon.
<img width="482" alt="screenshot 2018-12-28 at 14 03 25" src="https://user-images.githubusercontent.com/7622933/50516167-7d553e00-0aa9-11e9-9a77-84a97e8fef97.png">
